### PR TITLE
update vendored gengo

### PR DESCRIPTION
**Pull request by nikhita**
Need to fix defaulter-gen for https://github.com/kubernetes/kubernetes/pull/47263.
To pull in changes from https://github.com/kubernetes/gengo/pull/61.

**Release note**:

```release-note
NONE
```

/cc @sttts

### CODE REVIEW

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -415,7 +416,7 @@
 		},
 		{
 			"ImportPath": "github.com/codegangsta/negroni",
-			"Comment": "v0.1-62-g8d75e11",
+			"Comment": "v0.1.0-62-g8d75e11",
```
Comment (nikhita) [inline Godeps/Godeps.json]: @sttts ptal

File: hack/godep-save.sh [hunk 2]
```diff
@@ -30,6 +30,7 @@ REQUIRED_BINS=(
   "github.com/ugorji/go/codec/codecgen"
   "github.com/onsi/ginkgo/ginkgo"
   "github.com/jteeuwen/go-bindata/go-bindata"
+  "k8s.io/gengo/examples/..."
```
Comment (nikhita) [inline hack/godep-save.sh]: Added this. :+1:

Comment (nikhita) [inline hack/godep-save.sh]: Added this. :+1:

**Comment by nikhita** [comment]
cc @lavalamp for approval!

**Comment by cblecker** [comment]
verify is failing due to https://github.com/kubernetes/kubernetes/pull/48653

**Comment by nikhita** [comment]
Thanks, @cblecker.

**Comment by nikhita** [comment]
The tests pass locally though:

```
➜  kubernetes git:(defaulter-gen-vendor) ./hack/verify-staging-client-go.sh                                  132s 
Smoke testing client-go examples
Checking whether godeps are restored
Running staging/copy.sh
  creating the tmp directory
  copying client packages
  rewriting imports
  rearranging directory layout
  running gofmt
  remove black listed files
  running FAIL_ON_CHANGES
  /home/nikhita/code/kubernetes/src/k8s.io/kubernetes/staging/src/k8s.io/client-go up to date.
```

**Comment by nikhita** [comment]
Probably hitting https://github.com/kubernetes/kubernetes/issues/48675. 

Running ` ./hack/update-codegen.sh` did not update the auto-generated files at all places where it was necessary.

Removed the files that the CI complained of and regenerated them using make. They updated now.

**Comment by sttts** [comment]
@nikhita can you squash the last two commits?

**Comment by nikhita** [comment]
@sttts done.

**Comment by sttts** [comment]
Can you rebase?

**Comment by sttts** [comment]
@smarterclayton approved?

**Comment by nikhita** [comment]
I don't think this PR is needed anymore. gengo has already been updated in #48544. Closing.

**Pull request closed without merging**